src/DatePicker/DatePicker.js

```diff
@@ -171,17 +167,20 @@ class DatePicker extends React.Component {
     style: {},
   };
 
+
   static contextTypes = {
     muiTheme: React.PropTypes.object.isRequired,
   };
 
   state = {
-    date: undefined,
+    date: null,
+    dialogDate: null,
```

Comment:
@nathanmarks : You can see [here](https://github.com/callemall/material-ui/pull/3656/files) that dialog date was a part of the original code. some other PR must have removed it.